lib/rspec/core/configuration.rb

```diff
@@ -67,15 +67,17 @@ def self.define_reader(name)
       end
 
       # @private
-      def self.define_aliases(name, alias_name)
+      def self.define_alias(name, alias_name)
         alias_method alias_name, name
         alias_method "#{alias_name}=", "#{name}="
-        define_predicate_for alias_name
+        define_predicate alias_name
       end
 
       # @private
-      def self.define_predicate_for(*names)
-        names.each { |name| alias_method "#{name}?", name }
+      def self.define_predicate(name)
+        define_method "#{name}?" do
+          !!send(name)
```

Comment:
Does it have to be `send`, or `public_send` would be sufficient?

Reply:
`public_send` would probably be sufficient. It's not clear to me it would be an improvement here though, as we're calling ourselves. As much as we should replace `foo.bar` with `foo.public_send(:bar)`, I always replace `foo` with `send(:foo)`. If this was not meta-programming, you would not try to check if `name` was a private method or not. Even `self.foo` will work now, even if `foo` is private.

Reply:
We use `send` (or usually `__send__` because it is less likely to be overridden), this if fine :)

Reply:
Asking because for me `public_send` is a manifest "I'm not calling anything internal, just what you could do yourself, but provide a nicer interface for you".